contracts/Vault.vy

```diff
@@ -1733,23 +1708,21 @@ def report(gain: uint256, loss: uint256, _debtPayment: uint256) -> uint256:
     assert self.token.balanceOf(msg.sender) >= gain + _debtPayment
```

Comment:
Im thinking of a very deep edge case thats unlikely but nevertheless, it may be an attack.

MAlicious CustomCheck is activated by management that has a deposit balance in vault, 
when we check the numbers here its all good, but then malicious custom check when it gets called in health check proceeds to withdraw balance as reentrancy and strategy is in queue 0 index so it removes the balance reported before, stealing from vault profits or affecting vault accounting.

In that case the the assert invariant in 1708 is no longer valid after healthcheck since balanceOf changed. This is a highly unlikely and difficult attack to pull, but thinking if this assert should be either moved after healthcheck to enforce the balance.

Reply:
Can we add a small test to check if thats possible? Doggie mentioned that if the healthcheck call is VIEW it cannot make state changing calls after in calling stack. @pandadefi can we test with a mock custom check or malicious check that holds balance in the vault?

Reply:
I have added a check to make sure the call is a STATICCALL

https://github.com/yearn/yearn-vaults/pull/448/commits/7438b23cb6c065c8f0b28b4ef3008c78459a64b5

Reply:
I have also moved the balance check after the health check

https://github.com/yearn/yearn-vaults/pull/448/commits/5dabf5410730b23e29f4c7dc7cd2a49f514b4841